Sources/Segment/Plugins/SegmentDestination.swift

```diff
@@ -48,15 +49,23 @@ public class SegmentDestination: DestinationPlugin {
     private var apiHost: String? = nil
     
     @Atomic internal var eventCount: Int = 0
+    internal var flushAt: Int = 0
     internal var flushTimer: QueueTimer? = nil
     
     internal func initialSetup() {
         guard let analytics = self.analytics else { return }
         storage = analytics.storage
         httpClient = HTTPClient(analytics: analytics)
-        flushTimer = QueueTimer(interval: analytics.configuration.values.flushInterval) { [weak self] in
-            self?.flush()
+        
+        // flushInterval and flushAt can be modified post initialization
+        analytics.store.subscribe(self, initialState: true) { [weak self] (state: System) in
+            guard let self = self else { return }
+            self.flushTimer = QueueTimer(interval: state.configuration.values.flushInterval) { [weak self] in
```

Comment:
Not sure how this works in Swift but wouldn't it also have to cancel the previous timer?

Reply:
QueueTimer is a wrapper on a dispatch source timer.  Cancel is handled in dealloc.